src/Components/Web.JS/src/Platform/BootConfig.ts

```diff
@@ -1,13 +1,20 @@
+import { WebAssemblyBootResourceType } from './WebAssemblyStartOptions';
+
+type LoadBootResourceCallback = (type: WebAssemblyBootResourceType, name: string, defaultUri: string, integrity: string) =>
+  string | Promise<Response> | null | undefined;
+
 export class BootConfigResult {
   private constructor(public bootConfig: BootJsonData, public applicationEnvironment: string) {
   }
 
-  static async initAsync(environment?: string): Promise<BootConfigResult> {
-    const bootConfigResponse = await fetch('_framework/blazor.boot.json', {
-      method: 'GET',
-      credentials: 'include',
-      cache: 'no-cache'
-    });
+  static async initAsync(loadBootResource?: LoadBootResourceCallback, environment?: string): Promise<BootConfigResult> {
+    const loaderResponse = loadBootResource !== undefined ?
+      loadBootResource('manifest', 'blazor.boot.json', '_framework/blazor.boot.json', '') :
+      this.defaultLoadBlazorBootJson('_framework/blazor.boot.json');
+
+    const bootConfigResponse = loaderResponse instanceof Promise ?
+      await loaderResponse :
+      await BootConfigResult.defaultLoadBlazorBootJson(loaderResponse ?? '_framework/blazor.boot.json');
```

Comment:
Is there any distinction between `this.defaultLoadBlazorBootJson` (on line 13) and `BootConfigResult.defaultLoadBlazorBootJson` (on line 17)? I think they are the same thing. If so, to avoid confusion, it would be ideal to reference the function in the same way both times. Apologies if I'm missing some reason why it needs to vary.